src/frontend/src/components/Wallet.vue

```diff
@@ -20,13 +20,23 @@ import { mapState } from "vuex";
       (newValue: any, oldValue: any) => {
         if (newValue.length > oldValue.length) {
           this.$store.dispatch("ACTION_FETCH_WALLET_AMOUNT");
-
-          // Displays toast if transaction was recieved
-          const message = "New transaction received!";
-          this.$store.dispatch("ACTION_DISPLAY_TOAST", {
-            message: message,
-            type: "success",
-          });
+          const toSearch = []
+          let i: number;
+          for (i = oldValue.length; i < newValue.length; i++) {
```

Comment:
Suggestion: You could make this a while loop, and exit and display loop as soon as you see a reward/received message instead of going through all new transactions. However, we may never have a lot of new transactions so leaving it this was is fine too!

Reply:
Updated, please check new solution